dockerfiles/vcpkg/build.sh

```diff
@@ -0,0 +1,57 @@
+#!/bin/bash
+set -e
+
+SCRIPT_DIR="$( cd -- "$( dirname -- "${BASH_SOURCE[0]:-$0}"; )" &> /dev/null && pwd 2> /dev/null; )";
+
+export $(grep -v '^#' $SCRIPT_DIR/../../.env | xargs -d '\r' | xargs -d '\n') > /dev/null
+
+GITHUB_ACTOR="${GITHUB_ACTOR:-hpcc-systems}"
+GITHUB_TOKEN="${GITHUB_TOKEN:-none}"
+GITHUB_REF=$(git rev-parse --short=8 HEAD)
+cd vcpkg
+VCPKG_REF=$(git rev-parse --short=8 HEAD)
+cd ..
+DOCKER_USERNAME="${DOCKER_USERNAME:-hpccbuilds}"
+DOCKER_PASSWORD="${DOCKER_PASSWORD:-none}"
+
+echo "SCRIPT_DIR: $SCRIPT_DIR"
+echo "GITHUB_ACTOR: $GITHUB_ACTOR"
+echo "GITHUB_TOKEN: $GITHUB_TOKEN"
+echo "GITHUB_REF: $GITHUB_REF"
+echo "VCPKG_REF: $VCPKG_REF"
+echo "DOCKER_USERNAME: $DOCKER_USERNAME"
+echo "DOCKER_PASSWORD: $DOCKER_PASSWORD"
+
+docker login -u $DOCKER_USERNAME -p $DOCKER_PASSWORD
+
+function doBuild() {
+    docker build --progress plain --pull --rm -f "$SCRIPT_DIR/$1.dockerfile" \
+        -t build-$1:$GITHUB_REF \
+        -t build-$1:latest \
+        --build-arg VCPKG_REF=$VCPKG_REF \
+        "$SCRIPT_DIR/." 
+
+    docker run --rm --mount source="$(pwd)",target=/hpcc-dev/HPCC-Platform,type=bind,consistency=cached build-$1:$GITHUB_REF \
+        "cmake -S /hpcc-dev/HPCC-Platform -B /hpcc-dev/HPCC-Platform/build-$1 \${CMAKE_OPTIONS}"
+
+    docker run --rm --mount source="$(pwd)",target=/hpcc-dev/HPCC-Platform,type=bind,consistency=cached build-$1:$GITHUB_REF \
```

Comment:
minor: as per previous comments, could reuse same container afaics.

Reply:
Again I would need to see what that looks like...